### PR TITLE
Kronos as single service

### PR DESCRIPTION
**Pull request by tuhinsharma**
Gnosis, Softnet are made part of Kronos now. So there will be only a single service named Kronos. It will require some further iterations though.

### CODE REVIEW

**Comment by msrb** [comment]
Huge PR :) Since this is mostly refactoring of existing, already merged code, I'd say LGTM if the code works as expected. We can fix bugs as we go and merging this quickly will unblock us on CI/CD side. @miteshvp WDYT?

File: analytics_platform/kronos/deployment/rest_api.py [hunk 1]
```diff
@@ -47,11 +53,25 @@ def heart_beat():
...
 def train_and_save_kronos():
     app.logger.info("Submitting the training job")
+
     input_json = request.get_json()
     training_data_url = input_json.get("training_data_url")
+    fp_min_support_count = FP_MIN_SUPPORT_COUNT
+    fp_intent_topic_count_threshold = FP_INTENT_TOPIC_COUNT_THRESHOLD
+    fp_num_partition = FP_NUM_PARTITION
+    if "fp_min_support_count" in input_json:
+        fp_min_support_count = input_json.get("fp_min_support_count")
```
Comment (miteshvp) [inline analytics_platform/kronos/deployment/rest_api.py]: or
fp_min_support_count = input_json.get("fp_min_support_count", FP_MIN_SUPPORT_COUNT)

File: analytics_platform/kronos/deployment/rest_api.py [hunk 2]
```diff
@@ -47,11 +53,25 @@ def heart_beat():
...
+    if "fp_intent_topic_count_threshold" in input_json:
+        fp_intent_topic_count_threshold = input_json.get("fp_intent_topic_count_threshold")
```
Comment (miteshvp) [inline analytics_platform/kronos/deployment/rest_api.py]: same case as above

File: analytics_platform/kronos/deployment/rest_api.py [hunk 3]
```diff
@@ -47,11 +53,25 @@ def heart_beat():
...
+    if "fp_num_partition" in input_json:
+        fp_num_partition = input_json.get("fp_num_partition")
```
Comment (miteshvp) [inline analytics_platform/kronos/deployment/rest_api.py]: same case as above

**Comment by TuhinSharma** [comment]
Sure @miteshvp . I shall do those changes and update this PR.

**Comment by msrb** [comment]
OpenShift template references 3 S3 buckets:
- tusharma-kronos-data
- tusharma-gnosis-data
- tusharma-softnet-data

Are all 3 of them still needed? I am doing some changes to the template...

**Comment by TuhinSharma** [comment]
@msrb None of them are required now. I removed all of them in https://github.com/fabric8-analytics/fabric8-analytics-stack-analysis/pull/7/commits/ea6e5aec40be2f3b3053b3e6fcc1864a2b129702 . @miteshvp I did all changes as you suggested. Please check it once.

**Comment by msrb** [comment]
@TuhinSharma 

> None of them are required now. I removed all of them in ea6e5ae.

Thanks :) Just to double-check that I understand this correctly. Does that mean that Kronos service now doesn't store anything to any S3 bucket?

**Comment by tuhinsharma** [comment]
The kronos service accepts request json as follows:-
```
{
  "training_data_url": "s3://stack-analysis-data/python/github/",
  "fp_min_support_count" : 45,
  "fp_intent_topic_count_threshold" : 3,
  "fp_num_partition" : 12
}
```
so ```training_data_url``` is the s3 location which is passed in the REST call for training. The data is read from this location, model is trained and the model is stored in the same location. This doc has the information on it (https://docs.google.com/a/redhat.com/document/d/1AvvXuHNxTqVcKNtMr4kOUWchti01XbErQuwsNh54kpQ/edit?usp=sharing). But you shall find separate services Gnosis, Softnet and Kronos which was our previous approach. This will be modified soon and only one service should be there as Kronos in the api doc.

**Comment by miteshvp** [comment]
LGTM !

**Pull request merged**